examples/AvroSpecificWebApi/Controllers/ProducerController.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AvroSpecificWebApi.Entities;
+using Confluent.Kafka;
+using Confluent.SchemaRegistry;
+using Confluent.SchemaRegistry.Serdes;
+using Microsoft.AspNetCore.Mvc;
+
+namespace AvroSpecificWebApi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ProducerController : Controller
+    {
+        // Note: WebApi and MVC controller methods that use Confluent.Kafka MUST use the async. 
+        // Using .Result or .GetAwaiter().GetResult() will cause your application to hang. 
+        [HttpPost]
+        public async Task<string> Post([FromBody]Entities.Producer producerSettings)
+        {
+            var producerConfig = new ProducerConfig
+            {
+                BootstrapServers = producerSettings.BootstrapServers
+            };
+
+            var schemaRegistryConfig = new SchemaRegistryConfig
+            {
+                // Note: you can specify more than one schema registry url using the
+                // schema.registry.url property for redundancy (comma separated list). 
+                // The property name is not plural to follow the convention set by
+                // the Java implementation.
+                SchemaRegistryUrl = producerSettings.SchemaRegistryUrl,
+                // optional schema registry client properties:
+                SchemaRegistryRequestTimeoutMs = 5000,
+                SchemaRegistryMaxCachedSchemas = 10
+            };
+
+            var consumerConfig = new ConsumerConfig
+            {
+                BootstrapServers = producerSettings.BootstrapServers,
+                GroupId = "avro-specific-example-group"
+            };
+
+            var avroSerializerConfig = new AvroSerializerConfig
+            {
+                // optional Avro serializer properties:
+                BufferBytes = 100,
+                AutoRegisterSchemas = true
+            };
+
+            // Note: The User class in this project was generated using the Confluent fork of the avrogen.exe tool 
+            // (avaliable from: https://github.com/confluentinc/avro/tree/confluent-fork) which includes modifications
+            // that prevent namespace clashes with user namespaces that include the identifier 'Avro'. AvroSerializer
+            // and AvroDeserializer are also compatible with classes generated by the official avrogen.exe tool 
+            // (available from: https://github.com/apache/avro), with the above limitation.
+
+            using (var schemaRegistry = new CachedSchemaRegistryClient(schemaRegistryConfig))
+            using (var producer =
```

Comment:
creating a new producer for each request will be un-workably slow - there is a lot of overhead in establishing connections / querying API versions / getting metadata on startup. you should arrange things so there is one producer instance per application process, eg using application event hooks or a singleton class.

note that the kafka client doesn't work like a typical DB driver which typically manages a pool of open connections, giving you one of those when you request a new connection. the kafka client maintains a single open connection to each broker and makes maximal use of these connections (batching requests together etc).

Reply:
Done! I will cancel this PR, and submit a new one with all changed. Thank you!